src/main/java/org/fcrepo/upgrade/utils/F47ToF5UpgradeManager.java

```diff
@@ -298,14 +296,26 @@ private void upgradeRdfAndCreateHeaders(boolean isBinaryDescription, TemporalAcc
         LOGGER.debug("containerSubject={}", containerSubject);
     }
 
+    private String locateBinaryHeadersPrefixForVersionedBinary(final Path newLocation) {
+        //the idea here is translate a versioned metadata resource
+        //   from
+        //   path/to/binary/fcr%3Ametdata/fcr%3Aversions/20201105171804.ttl
+        //   to
+        //   path/to/binary/fcr%3Aversions/20201105171804
+        final var fileName = newLocation.getFileName().toString();
+        final var filenameWithoutExtension = fileName.substring(0,fileName.length()-TURTLE_EXTENSION.length());
```

Comment:
Probably less fragile to use [FilenameUtils](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FilenameUtils.html).

Reply:
Done.